okhttp-testing-support/src/main/kotlin/okhttp3/OkHttpClientTestRule.kt

```diff
@@ -173,7 +173,9 @@ class OkHttpClientTestRule : BeforeEachCallback, AfterEachCallback {
       // a test timeout failure.
       val waitTime = (entryTime + 1_000_000_000L - System.nanoTime())
       if (!queue.idleLatch().await(waitTime, TimeUnit.NANOSECONDS)) {
-        TaskRunner.INSTANCE.cancelAll()
+        synchronized (TaskRunner.INSTANCE) {
```

Comment:
The fail wasn't happening, because cancelAll fails.

Reply:
oooh good catch